ipfs/converter.go

```diff
@@ -104,3 +108,29 @@ func GetCID(desc ocispec.Descriptor) (string, error) {
 	}
 	return "", fmt.Errorf("no CID is recorded")
 }
+
+// GetIPFSAPIAddress get IPFS API URL from the specified IPFS repository.
+// If ipfsPath == "", then it's default is "~/.ipfs".
+// This is compatible to IPFS client behaviour: https://github.com/ipfs/go-ipfs-http-client/blob/171fcd55e3b743c38fb9d78a34a3a703ee0b5e89/api.go#L67-L82
+func GetIPFSAPIAddress(ipfsPath string) (string, error) {
+	if ipfsPath == "" {
+		ipfsPath = "~/.ipfs"
+	}
+	baseDir, err := homedir.Expand(ipfsPath)
+	if err != nil {
+		return "", err
+	}
+	api, err := os.ReadFile(filepath.Join(baseDir, "api"))
+	if err != nil {
+		return "", err
+	}
+	a, err := ma.NewMultiaddr(strings.TrimSpace(string(api)))
+	if err != nil {
+		return "", err
+	}
+	_, url, err := manet.DialArgs(a)
+	if err != nil {
+		return "", err
+	}
+	return "http://" + url, nil // TODO: support other protocols?
```

Comment:
No support for https?

Reply:
Is this a regression, or did we never support https?

Reply:
Fixed `ipfs/client/client.go` to allow other protocols. Stargz snapshotter still doesn't support https but I think this isn't a regression because when we used [`NewLocalApi` of `go-ipfs-http-client`](https://github.com/ipfs/go-ipfs-http-client/blob/171fcd55e3b743c38fb9d78a34a3a703ee0b5e89/api.go#L45), it didn't seem to support https as well (it seems that go-ipfs won't support https?: https://discuss.ipfs.tech/t/installing-ssl-certificate/388/2).